tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -65,11 +65,13 @@ namespace sm {
 
 FragmentMetadata::FragmentMetadata(
     StorageManager* storage_manager,
+    MemoryTracker* memory_tracker,
```

Comment:
Pass in the array here and use the tracker directly from there.

Reply:
I don't think that's a good idea. The fragment metadata can be loaded by `FragmentInfo`, which now does not require opening an array. Currently, there is no memory tracker for `FragmentInfo` similar to array, but I have noted it down to add one in a subsequent PR. In that case, a separate memory tracker will live in `FragmentInfo`. Hence the name change from `OpenArrayMemoryTracker` to `MemoryTracker`, as that class is applicable beyond arrays.

Reply:
You are right, I didn't think of the fragment info path. Ok let's push this out to another PR... This was my main reason to request changes. So I'll approve now...